### PR TITLE
Replace t.Fatal with require package in tests-e2e

### PR DESCRIPTION
**Pull request by AshwinSriram11**
## Description
Continuation to #6749, #6690, #6683, #6680 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Review by simonpasquier — APPROVED** [review]
Thanks, this is awesome!

**Comment by simonpasquier** [comment]
Could you rebase please?

File: test/e2e/prometheusagent_test.go [hunk 1]
```diff
@@ -160,20 +148,12 @@ func testPrometheusAgentStatusScale(t *testing.T) {
...
 	pAgent, err := framework.CreatePrometheusAgentAndWaitUntilReady(ctx, ns, pAgent)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	if pAgent.Status.Shards != 1 {
-		t.Fatalf("expected scale of 1 shard, got %d", pAgent.Status.Shards)
-	}
+	require.Equal(t, 1, pAgent.Status.Shards)
```
Comment (simonpasquier) [inline test/e2e/prometheusagent_test.go:153]:
```suggestion
	require.Equal(t, int32(1), pAgent.Status.Shards)
```

File: test/e2e/prometheusagent_test.go [hunk 2]
```diff
@@ -160,20 +148,12 @@ func testPrometheusAgentStatusScale(t *testing.T) {
...
 	pAgent, err = framework.ScalePrometheusAgentAndWaitUntilReady(ctx, name, ns, 2)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	if pAgent.Status.Shards != 2 {
-		t.Fatalf("expected scale of 2 shards, got %d", pAgent.Status.Shards)
-	}
+	require.Equal(t, 2, pAgent.Status.Shards)
```
Comment (simonpasquier) [inline test/e2e/prometheusagent_test.go:158]:
```suggestion
	require.Equal(t, int32(2), pAgent.Status.Shards)
```

**Comment by AshwinSriram11** [comment]
There are more files left in test/e2e left so I will do it in different PR. 

One of the files have like 273 appearances so do you want me to have a separate PR for just one file or multiple files in one PR? I am fine both ways. @simonpasquier

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Review by simonpasquier — APPROVED** [review]

**Review by slashpai — DISMISSED** [review]

**Review by simonpasquier — APPROVED** [review]

**Comment by ArthurSens** [comment]
Hi @AshwinSriram11, looks like there are some conflicts again 😕. Do you mind rebasing the PR?

**Comment by AshwinSriram11** [comment]
> Hi @AshwinSriram11, looks like there are some conflicts again 😕. Do you mind rebasing the PR?

I think I need to rebase it again after your PR #6804 gets merged.

**Comment by ArthurSens** [comment]
> > Hi @AshwinSriram11, looks like there are some conflicts again 😕. Do you mind rebasing the PR?
> 
> I think I need to rebase it again after your PR #6804 gets merged.

Correct, sorry for the trouble 😞

**Comment by AshwinSriram11** [comment]
> > > Hi @AshwinSriram11, looks like there are some conflicts again 😕. Do you mind rebasing the PR?
> > 
> > 
> > I think I need to rebase it again after your PR #6804 gets merged.
> 
> Correct, sorry for the trouble 😞

No problem 🙂

**Review by simonpasquier — APPROVED** [review]

**Review by slashpai — APPROVED** [review]

**Pull request merged**